### PR TITLE
feat(`provider`): `eth_getAccount` support

### PR DESCRIPTION
**Pull request by Evalir**
## Motivation

See https://t.me/ethers_rs/37923.

## Solution

Implement `eth_getAccount`.

Haven't added a test, as Anvil doesn't implement the method—needs an anvil companion PR

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/rpc-types/src/eth/account.rs [hunk 1]
```diff
@@ -41,6 +41,20 @@ pub struct EIP1186AccountProofResponse {
     pub storage_proof: Vec<EIP1186StorageProof>,
 }
 
+/// Response for `eth_getAccount`
+#[derive(Copy, Clone, Debug, Default, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
```
Comment (prestwich) [inline crates/rpc-types/src/eth/account.rs:46]: do we have an Account struct in alloy-consensus? if not i would prefer to push this down and ensure it has RLP encoding

File: crates/provider/src/provider/trait.rs [hunk 2]
```diff
@@ -574,6 +574,16 @@ pub trait Provider<T: Transport + Clone = BoxTransport, N: Network = Ethereum>:
         self.client().request("eth_getLogs", (filter,)).await
     }
 
+    /// Retrieves account information ([AccountResponse]) for the given [Address] at the particular
+    /// [BlockId].
+    async fn get_account(
+        &self,
+        address: Address,
+        block: BlockId,
```
Comment (mattsse) [inline crates/provider/src/provider/trait.rs:582]: can we reuse the new call struct here and only accept address?

File: crates/rpc-types/src/eth/account.rs (hunk 1 above)
Comment (mattsse) [reply crates/rpc-types/src/eth/account.rs:46]: yeah we should either move this to consensus or eips, maybe eip1186 because more or less related

**Review by mattsse — CHANGES_REQUESTED** [review]
one note here, the eth api proposal for this was closed because this no longer makes sense post verkle, but rpc providers currently support this, so reasonable to add

**Review by mattsse — APPROVED** [review]
lgtm

**Review by prestwich — CHANGES_REQUESTED** [review]
still needs struct pushed down

**Comment by mattsse** [comment]
yeah, I realized we don't have eip1186 in the eips yet, so I think we should move them separately.

**Comment by prestwich** [comment]
the account struct should be in consensus, as it's an member of the state trie, no?

**Comment by mattsse** [comment]
where do you want me to put this, consensus/account.rs ?

**Comment by prestwich** [comment]
that'd be great. I can get to it in a couple hours if you don't. if you want to merge this before adding the RLP impl, please open an issue to follow up 🫡

**Comment by mattsse** [comment]
this would introduce alloy-consensus dep in provider, not sure if we want this

**Comment by prestwich** [comment]
don't we already have that indirectly via receipts and txenvelopes?

**Comment by mattsse** [comment]
ah we indeed have this already

**Review by prestwich — APPROVED** [review]

**Pull request merged**